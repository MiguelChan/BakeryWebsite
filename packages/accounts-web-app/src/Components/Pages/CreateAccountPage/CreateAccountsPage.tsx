import React from 'react';
import {
  Account,
} from '@mgl/shared-components';
import { AccountsAppContext, ApplicationContext } from '../../../Context';
import { CreateAccountForm } from '../../Constructed/CreateAccountForm/CreateAccountForm';
import { SimpleAppTemplate } from '../../Templates';
import { LoadingDialog } from '../../Blocks';
import { Alert, AlertTitle } from '@mui/material';
import { Redirect } from 'react-router-dom';

export const CreateAccountsPage: React.FunctionComponent = () => {

  const {
    useCreateAccount,
  } = React.useContext<ApplicationContext>(AccountsAppContext);
  const useCreateAccountState = useCreateAccount();

  React.useEffect(() => {
    
  }, [useCreateAccountState]);

  const onSubmitAccount = (account: Account): void => {
    useCreateAccountState.requestCreateAccount(account);
  }

  const renderErrorAlert = (): React.ReactElement => (
    <Alert severity='error'>
      <AlertTitle>Un error ha ocurrido</AlertTitle>
      {useCreateAccountState.errorMessage}
    </Alert>
  );

  const renderBody = (props: any): React.ReactElement => {
    const {
      errorMessage,
      isLoading,
      isAccountCreated,
    } = useCreateAccountState;
    return (
      <>
        {isLoading && <LoadingDialog onClose={(): void => {}}/>}
        {errorMessage && renderErrorAlert()}
        {isAccountCreated && <Redirect to='/accounts' />}
        <CreateAccountForm 
          {...props}
          onSubmitAccount={onSubmitAccount}
        />
      </>
    );
  }

  return (
    <SimpleAppTemplate
      renderBody={renderBody}
    />
  );
};
