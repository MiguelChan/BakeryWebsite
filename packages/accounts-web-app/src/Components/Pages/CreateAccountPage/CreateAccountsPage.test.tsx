import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { CreateAccountsPage } from './CreateAccountsPage';
import { 
  Loading,
  Primary,
  WithErrorMessage,
} from './CreateAccountsPage.stories';
import { AccountsAppContext } from '../../../Context';

describe('CreateAccountsPage', () => {
  const setupComponent = ({appContext}): RenderResult => {
    const Component = (
      <AccountsAppContext.Provider value={appContext}>
        <CreateAccountsPage />
      </AccountsAppContext.Provider>
    );
    return render(Component);
  }

  it('Should display the Component', () => {
    const renderResult = setupComponent(Primary.args);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should display the Loading Dialog', () => {
    const renderResult = setupComponent(Loading.args);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should display the Error message', () => {
    const renderResult = setupComponent(WithErrorMessage.args);
    expect(renderResult.container).toMatchSnapshot();
  });
});