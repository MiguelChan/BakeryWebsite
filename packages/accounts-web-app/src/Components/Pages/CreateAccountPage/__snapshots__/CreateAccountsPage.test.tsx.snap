// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountsPage Should display the Component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h5 css-1hxkzy9-MuiTypography-root"
      >
        Creacion de Cuenta
      </h4>
      <form
        id="create-account"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="title"
                id="title-label"
              >
                Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="AccountTitleId"
                  id="title"
                  name="title"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-dhh0px"
                  >
                    <span>
                      Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                Tipo de Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="accountType"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="accountType"
                  role="button"
                  tabindex="0"
                >
                  Capital
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="accountType"
                  tabindex="-1"
                  value="Capital"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Tipo de Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Subcuentas
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Id de Subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Nombre de subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              />
            </table>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-dlyzdu-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              data-testid="CreateAccountId"
              form="create-account"
              tabindex="0"
              type="submit"
            >
              Crear cuenta
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-mrzqee-MuiButtonBase-root-MuiButton-root"
              data-testid="AddSubAccountId"
              tabindex="0"
              type="button"
            >
              Agregar Subcuenta
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`CreateAccountsPage Should display the Error message 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-145dpyk-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
        >
          Un error ha ocurrido
        </div>
        El sistema se ha caido y ya no funciona
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h5 css-1hxkzy9-MuiTypography-root"
      >
        Creacion de Cuenta
      </h4>
      <form
        id="create-account"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="title"
                id="title-label"
              >
                Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="AccountTitleId"
                  id="title"
                  name="title"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-dhh0px"
                  >
                    <span>
                      Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                Tipo de Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="accountType"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="accountType"
                  role="button"
                  tabindex="0"
                >
                  Capital
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="accountType"
                  tabindex="-1"
                  value="Capital"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Tipo de Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Subcuentas
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Id de Subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Nombre de subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              />
            </table>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-dlyzdu-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              data-testid="CreateAccountId"
              form="create-account"
              tabindex="0"
              type="submit"
            >
              Crear cuenta
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-mrzqee-MuiButtonBase-root-MuiButton-root"
              data-testid="AddSubAccountId"
              tabindex="0"
              type="button"
            >
              Agregar Subcuenta
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`CreateAccountsPage Should display the Loading Dialog 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h5 css-1hxkzy9-MuiTypography-root"
      >
        Creacion de Cuenta
      </h4>
      <form
        id="create-account"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="title"
                id="title-label"
              >
                Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="AccountTitleId"
                  id="title"
                  name="title"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-dhh0px"
                  >
                    <span>
                      Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                Tipo de Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="accountType"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="accountType"
                  role="button"
                  tabindex="0"
                >
                  Capital
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="accountType"
                  tabindex="-1"
                  value="Capital"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Tipo de Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Subcuentas
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Id de Subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Nombre de subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              />
            </table>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-dlyzdu-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              data-testid="CreateAccountId"
              form="create-account"
              tabindex="0"
              type="submit"
            >
              Crear cuenta
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-mrzqee-MuiButtonBase-root-MuiButton-root"
              data-testid="AddSubAccountId"
              tabindex="0"
              type="button"
            >
              Agregar Subcuenta
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
